packages/ui-extensions/src/types/Button.ts

```diff
@@ -0,0 +1,58 @@
+export interface BaseButtonProps {
+  /**
+   * Destination to link to, renders a Link
+   */
+  to?: string;
+  /**
+   * The type of button that will be rendered.
+   *
+   * `primary`: button used for main actions or green-path. Ex: Continue to next step, Discount field
+   * `secondary`: button used for secondary actions not blocking user progress. Ex: Download Shop app
+   * `plain`: Renders a button that visually looks like a Link
+   * @default 'primary'*/
+  kind?: 'primary' | 'secondary' | 'plain';
```

Comment:
IMO this prop is too design-system specific, there's no guarantee checkout will be able to frame the different button types this way

Reply:
@lemonmade You liked this approach in April 😂 !  https://github.com/Shopify/argo-rfcs/issues/2#issuecomment-824058184. That said, I know things I things evolve. @jffortier addresses these concerns below. If these appearance is still editable by the merchant, does this sit better with you? On the Admin side, this maps pretty well with Polaris.  While they don't have `secondary`, but we can map this to the non-primary kind.

Reply:
Hmm well the evidence is pretty overwhelming :P I think it will be pretty difficult to have a shared set of button types that are common across all our apps, but it seems like no one else is super worried about it so 👍

Reply:
I'd at least remove the `@default` here though — I don't think this is the default we will want within checkout in the end, and in general defaults seem like a pretty host-specific concern.